### PR TITLE
Initial documentation action

### PR DESCRIPTION
**Pull request by BlockSprintZIf**

### CODE REVIEW

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: CI
```
Comment (mloubout) [inline .github/workflows/documentation.yml]: documentation

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+on: [release]
```
Comment (mloubout) [inline .github/workflows/documentation.yml]: push to master maybe? I think the doc should be up to date with GitHub version, @FabioLuporini @ggorman ?

File: .github/workflows/documentation.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Run documentation script
```
Comment (mloubout) [inline .github/workflows/documentation.yml]: Copy paster leftover?

File: .github/workflows/documentation.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      run: echo Hello, world Documentation!
+    - name: Run a multi-line script
```
Comment (mloubout) [inline .github/workflows/documentation.yml]: Better name would be good

File: .github/workflows/documentation.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      run: |
+        echo Add other actions to build,
+        echo test, and deploy your project.
```
Comment (mloubout) [inline .github/workflows/documentation.yml]: These two echo are maybe not necessary

**Comment by navjotk** [comment]
Unsurprisingly, we are not the first people to try to use doctr with Github Actions: https://github.com/drdoctr/doctr/issues/351

**Comment by navjotk** [comment]
Addresses https://github.com/devitocodes/devito/issues/1057

File: .github/workflows/documentation.yml (hunk 2 above)
Comment (mloubout) [reply .github/workflows/documentation.yml]: on:
   # Trigger the workflow on push or pull request,
   # but only for the master branch
   push:
     branches:
       - master

**Comment by mloubout** [comment]
the `make html` part is missing

**Comment by FabioLuporini** [comment]
how can we test whether this actually works ?

**Comment by navjotk** [comment]
I'm testing this here: https://github.com/navjotk/devito/pull/1

File: .github/workflows/documentation.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        python-version: 3.7
+    - name: Run documentation script
+      run: |
```
Comment (mloubout) [inline .github/workflows/documentation.yml]: Would keep the actions syntax. organization:

```
- name: install dependencies
  run: |
    python -m pip install --upgrade pip 
    pip install sphinx doctr

- name: build
  run: make html
  working_directory: docs

- name: Deploy documentation
   run: doctr deploy . --built-docs docs/_build/html --key-path .deploy_key.enc
```

Comment (navjotk) [reply .github/workflows/documentation.yml]: I think your comment is outdated

Comment (mloubout) [reply .github/workflows/documentation.yml]: It's not, pip and make in same part, and actions uses `working_directory` not `cd`.  Also `actions-gh-pages` doesn't to use doctr so doesn't need to be installed.

Comment (mloubout) [reply .github/workflows/documentation.yml]: Sorry this one was outdated yes

**Pull request merged**